source/route_auth.js

```diff
@@ -12,27 +12,27 @@ angular.module( "vokal.RouteAuth", [ "LocalStorageModule" ] )
         redirectPath = path;
     };
 
-    this.$get = [ "$q", "$location", "$rootScope", "localStorageService",
+    this.$get = [ "$q", "$location", "$rootScope",
 
-        function ( $q, $location, $rootScope, localStorageService )
+        function ( $q, $location, $rootScope )
         {
             var service = {
-
+                storageType: localStorage,
```

Comment:
I thought about that when exposing it this way. As you pointed out, there are several things that could happen if there was a method to transition from one storage type to the other
1. Transition nothing, like it currently does
1. Copy to the new storage location and leave the old location as is
1. Copy to the new storage location and clear the old location

I'm guessing that most people would want/expect the 3rd behavior to happen if it is named something like `swapStorage( medium )` where `medium` is assumed to be `localStorage` or `sessionStorage` but could really be anything that just implements `setItem` and `getItem`. If `swapStorage( medium )` was added, I'd make `storageType` completely internal.